pkg/deps/config/suppliers.go

```diff
@@ -95,17 +94,29 @@ func NewSupplyBlockClientFn(queryHost string) SupplierFn {
 // NewSupplyQueryClientContextFn returns a function with constructs a ClientContext
 // instance with the given cmd and returns a new depinject.Config which is
 // supplied with the given deps and the new ClientContext.
-func NewSupplyQueryClientContextFn(pocketQueryNodeURL string) SupplierFn {
+func NewSupplyQueryClientContextFn(
+	pocketQueryNodeGRPCURL *url.URL,
+	pocketQueryNodeGRPCInsecure bool,
+) SupplierFn {
 	return func(_ context.Context,
 		deps depinject.Config,
 		cmd *cobra.Command,
 	) (depinject.Config, error) {
 		// Temporarily store the flag's current value
-		tmp := cosmosflags.FlagNode
+		tmp := cosmosflags.FlagGRPC
```

Comment:
```suggestion
		tmpGRPC := cosmosflags.FlagGRPC
		tmpINSECURE := cosmosflags.FlagGRPCInsecure
```
And restore it afterwards just the same